### ISSUE TITLE
Clusters that use the same subdomain cannot communicate with each other

### ISSUE THREAD

**Issue by csrwng** [openshift/installer#1136]
# Version 

(All)
```console
$ openshift-install version
openshift-install unreleased-master-145-ge26740933aff029a4ff1c7884fe7a4dee06482b6
```

# Platform:

aws

# What happened?

Install more than one cluster using the same DNS subdomain (ie. `dev.openshift.com`). You will get different URLs for api access. For instance `cluster1-api.dev.openshift.com` and `cluster2-api.dev.openshift.com`.

Attempting to access the API (or any application for that matter) that is hosted on cluster2 from a pod/machine in cluster1  will fail with a DNS lookup error. Same in the opposite direction. Note that this is not an issue for a client that lives outside of AWS.

# What you expected to happen?

That you can access either cluster's public endpoints from anywhere.

# How to reproduce it (as minimally and precisely as possible)?

Install 2 clusters as described above using the same DNS subdomain.

# Anything else we need to know?

This is a limitation of AWS and how we setup separate private zones for each cluster. DNS lookups by machines in one cluster will try to use the private DNS zone for that cluster, which knows nothing about machines that live in a different cluster. 

We should at the very least document that when creating a new cluster, to use a separate public zone for each cluster. Even better, when installing on AWS, the installer could create a separate public zone delegated from the main public zone. (for example if I say use dev.openshift.com, the installer could create cluster1.dev.openshift.com and use that).

**Comment by crawford** [comment]
Split horizon DNS is really hard. I think we should probably just include the cluster name in the base domain. That way, each cluster gets its own zone and there is no name collision. This is going to be extremely disruptive, so we should make the change ASAP.

**Comment by wking** [comment]
> I think we should probably just include the cluster name in the base domain.

Can we deprecate "cluster name" as a thing, and just have "cluster ID" and "cluster domain"?

**Comment by wking** [comment]
CC @dgoodwin, @vrutkovs, since they'd probably care about shuffling the config schema.

**Comment by vrutkovs** [comment]
Including cluster name in DNS (cluster1.dev.openshift.org) seems to be a good solution, its already used by etcd discovery afaik

**Comment by dgoodwin** [comment]
The cluster name is still used in naming entities in AWS, probably a lot of other places as well, parsing it out of the domain is doable just a little odd, and we'd have to hope users do it correctly. It's an easy change for us to switch to this though so if that's the way you wanna go no problem just tag us on the PR.

We could also just assume to combine name+basedomain when creating the DNS zone, which would mean no config format changes, and the config feels a little more clear to me at least, but I can see that being debatable.

**Comment by wking** [comment]
> The cluster name is still used in naming entities in AWS, probably a lot of other places as well, parsing it out of the domain is doable just a little odd,

I think we'd want to drop that, and just use the full cluster domain for name tags.  This would help with #762.

**Comment by dgoodwin** [comment]
That wouldn't be ideal for credentials, I have 64 chars to play with for cluster name, credential name, and random suffix. Might still be workable though but it's getting pretty tight by that point.

**Comment by smarterclayton** [comment]
I agree that having base + cluster name should go away.

**Comment by smarterclayton** [comment]
I hit this while trying to prototype GCP - they don't really support split zone yet completely so this was really annoying.

**Comment by wking** [comment]
> That wouldn't be ideal for credentials, I have 64 chars to play with for user names...

So include as much as you can of the 36-character cluster ID to get a good shot at uniqueness, and leave the full cluster domain to a tag?  Then you can drop the random suffix as well.

**Comment by crawford** [comment]
Fixed by #1169.
